Update Compose and others

djnotes opened the pull request:
Compose to 1.3.0
Filament to  1.28.2
Android Gradle plugin to 8.0.0-alpha07
Kotlin Gradle plugin to 1.7.10
Gradle to 7.5
In Viewer.kt, change the signature of onViewAttachedToWindow and onViewDetachedFromWindow to use android.view.View rather than android.view.View? because the former didn't compile with error "x does not implement y"
Other minor library updates

djnotes committed:
Update Compose and others

Compose to 1.3.0
Filament to  1.28.2
Android Gradle plugin to 8.0.0-alpha07
Kotlin Gradle plugin to 1.7.10
Gradle to 7.5
In Viewer.kt, change the signature of onViewAttachedToWindow and
onViewDetachedFromWindow to use android.view.View rather than
android.view.View? because the former didn't compile with error "x does
not implement y"
Other minor library updates

romainguy commented on .gitattributes:
```diff
@@ -0,0 +1 @@
+* text=auto
```
Please don't change the .git files

romainguy commented on .gitignore:
```diff
@@ -15,3 +15,4 @@
 .externalNativeBuild
 .cxx
 local.properties
+filament/bin/matc.exe
```
Keep next to the other filament/bin

romainguy commented on app/build.gradle:
```diff
@@ -55,27 +56,27 @@ android {
-    implementation "androidx.core:core-ktx:1.6.0"
-    implementation 'androidx.appcompat:appcompat:1.4.0-beta01'
+    implementation "androidx.core:core-ktx:1.9.0"
+    implementation 'androidx.appcompat:appcompat:1.7.0-alpha01'
```
Why use an alpha here?

romainguy commented on build.gradle:
```diff
@@ -1,16 +1,16 @@
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.3'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.31"
+        classpath 'com.android.tools.build:gradle:8.0.0-alpha07'
```
Let's not depend on previews of Android Studio

djnotes committed:
Update Compose and other libraries

Update:
  Android gradle plugin => 7.2.2
  Compose => 1.3.0
  appcompat => 1.5.1
  kotlinx-collections-immutable-jvm => 0.3.5

djnotes commented:
Rolled things back to stable and other stuff

romainguy commented on .idea/codeStyles/Project.xml:
```diff
@@ -0,0 +1,123 @@
+<component name="ProjectCodeStyleConfiguration">
```
Please nuke this file. It's changing the code style settings for this project

romainguy reviewed (approved):
Thank you!
